phc/util/api_cache.py

```diff
@@ -0,0 +1,69 @@
+import hashlib
+import json
+from pathlib import Path
+from typing import Callable
+
+import pandas as pd
+
+from phc.util.csv_writer import CSVWriter
+
+DIR = "~/_phc/api-cache"
+
+
+class APICache:
+    @staticmethod
+    def filename_for_fhir_dsl(query: dict):
+        "Descriptive filename with hash of query for easy retrieval"
+        components = [
+            "fhir",
+            "dsl",
+            *[d.get("table", "") for d in query.get("from", [])],
+            f"{len(query.get('columns', []))}col"
+            if isinstance(query.get("columns"), list)
+            else "",
+            "where" if query.get("where") else "",
+            hashlib.sha256(json.dumps(query).encode("utf-8")).hexdigest()[0:8],
```

Comment:
This is really hard to read since the some statements wrap multiple lines. Mind breaking this up into something that is easier to understand?